Updates to enable building and running on macOS 15.5 and Xcode 16.3

nickhart opened the pull request:
Out of the box the Code.org dashboard does not build for users running macOS 15.5 and Xcode 16.3. The older version of Ruby 3.0.5 conflicts with the latest version of Clang and requires patching or updating to a newer version.

In addition, `eyes_selenium` 4.x does not compile with Ruby 3.3.4, so it is necessary to update it to 6.9.

However, I didn't want to introduce any breaking changes by committing changes to either `.ruby-version` or `Gemfile` so I leave it to the developer to make those changes locally.

Changes in this PR:
* Changed lib/cdo/global_edition.rb to use `YAML.safe_load` to safely handle aliases (required for Ruby >= 3.1)
* Changed lib/cdo/yaml.rb to use the `YAML_ENABLE_ALIAS` env var
* Made some updates to SETUP.md:
** Added additional instructions for developers on modern macOS systems to get their build working.
** Added information on setting up MinIO to help external contributors use the S3 emulator.
** Elaborated the DB charset instructions to hopefully make it more clear.
** Fixed a typo.

## Testing story

Unit tests pass (`bundle exec rake test:all`) except for: `test/unit/maker/boards/microBit/MicroBitBoardTest.js`

## Deployment strategy

For now this may only be useful for devs running macOS 15.5 and Xcode 16.3 locally. I don't know that this will deploy cleanly with Ruby 3.0.5.

## Follow-up work

Validate these changes with an Ubuntu or older macOS system and Ruby 3.0.5? Officially support Ruby 3.3.4? 

## Privacy

  1.	Does this change involve the collection, use, or sharing of new Personal Data? **NO**
  2.	Does this change involve a new or changed use or sharing of existing Personal Data? **NO**

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

wilkie commented:
Hi! This is wonderful to see. I'm happy that only a few changes were really necessary and, yeah, the Mac experience is not as ideal as one would hope. You caught us at an interesting moment. We were deploying a ruby update to 3.1 (we go about it quite... I'd put it as 'carefully') so some of this might be moot.

I do like the updates to the specific documentation for external contributors. Some of this seems to be duplicating the content from `docker/developers/README.md` which we could probably link to in this part of our `SETUP.md` document to make things clearer. It's very nice that you are making this work while only using MinIO and not a dockerized database, etc.

I think the directions I would list instead would be to try (and ensure it is `run` and not `up` to see the output):

```
docker compose run s3-services
```

Which should output what to add to the `locals.yml` file.

In theory, you don't need to `-f <path to docker/developers/file.yml` and you should be able to just directly run `docker compose up minio` from the root. But I think the `s3-services` script is the friendlier option since it repeats the necessary documentation.

----

Long-winded reply!! Sorry, but you've volunteered a little bit as a test subject :) The things I would like to know are:

* How does the ruby 3.1 update change the situation?
* Does `docker compose run s3-services` work in your use-case?

nickhart commented:
Thanks, and sorry for the slow reply! I updated my branch to use Ruby 3.1.6 and it seems to work fine. 

`docker compose run  s3-services` also works and displays the following:

```
[+] Creating 1/1
 ✔ Container code-dot-org-minio-1  Created                                                                                                          0.0s 
[+] Running 1/1
 ✔ Container code-dot-org-minio-1  Started                                                                                                          0.1s 
S3 Emulation Services Started (MinIO)

To see and interact with the buckets:
http://localhost:3001

Username: local-development
Password: allstudents

To shut down the S3 service, use:
docker compose stop minio

You need to add the following to locals.yml to have your server use these services:

aws_s3_endpoint: http://localhost:9000
aws_s3_access_key_id: local-development
aws_s3_secret_access_key: allstudents
aws_s3_emulated: true
```

Those variables are the ones documented in [`locals.yml.default`](https://github.com/code-dot-org/code-dot-org/blob/a80834c6f35dab41c0be4bc8932d53f3d548d47f/locals.yml.default#L73) and what I added to the external contributors documentation in SETUP.md, although I suppose it's redundant and means if this ever changes it has to be updated in the `locals.yml.default` **and** `SETUP.md`.

I'm happy to do some tweaking of the docs... or just hold off for now since it sounds like Code.org is in the middle of doing some of this work anyway. I'm also happy to be a guinea pig if you need an external macOS tester.
